sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -848,9 +826,11 @@ case class ContextChecks(
       check.cudf.tagExprParam(meta, children(i), check.name, willNotWork)
     }
     if (repeatingParamCheck.isEmpty) {
-      assert(fixedChecks.length == children.length,
-        s"${expr.getClass.getSimpleName} expected ${fixedChecks.length} but " +
+      if (!meta.isChildExprsCountDynamic) {
+        assert(fixedChecks.length == children.length,
+          s"${expr.getClass.getSimpleName} expected ${fixedChecks.length} but " +
             s"found ${children.length}")
+      }
```

Comment:
Any check is required for the expression with dynamic children ?

Reply:
No, the `replacement` is unknown/inderministic. E.g.:
`between` is replaced by  `And(GreaterThanOrEqual(ref, lower), LessThanOrEqual(ref, upper)`; 
`StructToJson` is `replaced by Invoke(Literal(StructToJsonEvaluator), evaluate, string_type, arguments)`
`InvokeExprMeta` is responsible to do all the checking case by case.